File: src/main/java/com/wave/hextractor/util/GuiUtils.java
package com.wave.hextractor.util;

import javax.swing.*;

/**
 * Class for gui utilities.
 */
public class GuiUtils {

	/**
	 * Instantiates a new gui utils.
	 */
	private GuiUtils() {
	}

	/**
	 * Shows a confirmation dialog for the title and message passed.
	 * @return true if accepted.
	 */
	public static boolean confirmActionAlert(String title, String message) {
		return JOptionPane.showConfirmDialog(null, message, title,
				JOptionPane.YES_NO_OPTION, JOptionPane.QUESTION_MESSAGE) == JOptionPane.YES_OPTION;
	}
}

File: src/main/kotlin/com/wave/hextractor/util/GuiUtils.kt
package com.wave.hextractor.util

import javax.swing.JOptionPane

/**
 * Class for gui utilities.
 */
object GuiUtils {
    /**
     * Shows a confirmation dialog for the title and message passed.
     * @return true if accepted.
     */
    fun confirmActionAlert(title: String?, message: String?): Boolean {
        return JOptionPane.showConfirmDialog(
            null,
            message,
            title,
            JOptionPane.YES_NO_OPTION,
            JOptionPane.QUESTION_MESSAGE
        ) == JOptionPane.YES_OPTION
    }
}
